### PR TITLE
functors task 1 rudnichenko

### PR DESCRIPTION
**Pull request by CatBehemotth**

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -1,13 +1,22 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <iostream>
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:4]: _please don't leave debugging things_

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    size_t state {static_cast<size_t>(BeerBrand::None)};
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:12]: _consider more descriptive name_
why not BeerBrand type?

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -40,9 +51,26 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
-    {
+    static Cocktail mix(AlcoholDrink lhs, NonAlcoholDrink rhs)
+    {       
+            if (AlcoholDrink::Gin == lhs && NonAlcoholDrink::LimeJuice == rhs)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:56]: consider switch here

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (CatBehemotth) [reply functors task (1)/student/include/impl.h]: Done

File: functors task (1)/student/include/impl.h (hunk 2 above)
Comment (CatBehemotth) [reply functors task (1)/student/include/impl.h]: Done

File: functors task (1)/student/include/impl.h (hunk 3 above)
Comment (CatBehemotth) [reply functors task (1)/student/include/impl.h]: Done

File: functors task (1)/student/include/impl.h [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
-        return static_cast<BeerBrand>(state++);
+            beerBrandType = BeerBrand::HoeGaarden;
+        }    
+           
+        return beerBrandType;
     }
+
+private:
+
+ BeerBrand beerBrandType {BeerBrand::None};
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:23]: _please fix formatting_

File: functors task (1)/student/include/impl.h [hunk 5]
```diff
@@ -51,25 +55,33 @@ struct MixingPolicy
...
-                return Cocktail::Oops;
-            }      
-        
+                if (NonAlcoholDrink::SevenUp == nonAlcoholDrink)
+                {
+                    return Cocktail::SevenPlusSeven;
+                }
+            }
+            break;
+            default:
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:81]: _add return in default section_

**Review by smay1613 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**